Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+                    //{ fileID: 11500000, guid: 83d9acc7968244a8886f3af591305bcb, type: 3}
+
+                    await writer.WriteLineAsync(line);
+                }
+            }
+        }
+
+        private static bool IsYamlFile(string filePath)
+        {
+            string extension = Path.GetExtension(filePath);
+            if (ExcludedYamlAssetExtensions.Contains(extension))
+            {
+                return false;
+            }
+
+            using (StreamReader reader = new StreamReader(filePath))
+            {
+                return reader.ReadLine().StartsWith(YamlPrefix);
+            }
+        }
+
+        private static Dictionary<string, ClassInformation> ProcessScripts(string[] allFilePaths)
+        {
+            int lengthOfPrefix = Application.dataPath.IndexOf("Assets");
+
+            Dictionary<string, ClassInformation> toReturn = new Dictionary<string, ClassInformation>();
+
+            foreach (string filePath in allFilePaths)
+            {
+                if (Path.GetExtension(filePath) == ".cs")
+                {
+                    MonoScript monoScript = AssetDatabase.LoadAssetAtPath<MonoScript>(filePath.Substring(lengthOfPrefix));
+                    if (AssetDatabase.TryGetGUIDAndLocalFileIdentifier(monoScript, out string guid, out long fileId))
+                    {
+                        Type type = monoScript.GetClass();
+                        if (type != null)
+                        {
+                            toReturn.Add(type.FullName, new ClassInformation() { Name = type.Name, Namespace = type.Namespace, FileId = fileId, Guid = guid });
+                        }
+                        else
+                        {
+                            nonClassDictionary.Add(guid, Path.GetFileName(filePath));
+                            Debug.LogWarning($"Found script that we can't get type from: {monoScript.name}");
+                        }
+                    }
+                }
+            }
+
+            return toReturn;
+        }
+
+        private static Dictionary<string, ClassInformation> ProcessCompiledDLLs(string temporaryDirectoryName, string outputDirectory, Dictionary<string, string> asmDefMappings)
+        {
+            Assembly[] dlls = CompilationPipeline.GetAssemblies();
+
+            string tmpDirPath = Path.Combine(Application.dataPath, temporaryDirectoryName);
+            if (Directory.Exists(tmpDirPath))
+            {
+                Directory.Delete(tmpDirPath);
+            }
+
+            Directory.CreateDirectory(tmpDirPath);
+
+            try
+            {
+                foreach (Assembly dll in dlls)
+                {
+                    if (dll.name.Contains("MixedReality"))
+                    {
+                        File.Copy(dll.outputPath, Path.Combine(tmpDirPath, $"{dll.name}.dll"), true);
+                    }
+                }
+
+                // Load these directories
+                // TODO maybe we don't need to
```

Comment:
>// TODO maybe we don't need to [](start = 16, length = 30)

NIt: Should we just delete this TODO?

Reply:
Let me try without it :)

---
In reply to: [277834891](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277834891) [](ancestors = 277834891)

Reply:
Will remove Todo keep the refresh.

---
In reply to: [277835987](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277835987) [](ancestors = 277835987,277834891)